static/js/datatables_config.js

```diff
@@ -1,103 +1,98 @@
-const gen_datatables_default_config = (table_selector) => {
+const gen_datatables_config = (overwrites) => {
   const column_selector = (idx, data, node) => {
-	// https://datatables.net/forums/discussion/42192/exporting-data-with-buttons-and-responsive-extensions-controlled-by-column-visibility
-	return $(table_selector).DataTable().column( idx ).visible();
-  };
-  return {
-	dom: 'Bfrtipl',
-	responsive: true,
-	colReorder: true,
-	deferRender: true,
-	createdRow: (row) => {
-	  $(row).find('[data-toggle="popover"]').popover();
-	},
-	columnDefs: [
-	  {
-		targets: 'datatables-points',
-		render: (data, type, row) => {
-			if(type === 'display') {
-			  return data;
-			} else {
-				const div = document.createElement("div");
-				div.innerHTML = data;
-				const percent = div.innerText;
-			  if(type === 'export') {
-				return percent;
-			  } else {
-				return parseFloat(percent.replace('%', ''));
-			  }
-			}
-		  }
-	  },
-	  {
-		targets: 'datatables-status',
-		render: (data, type, row) => {
-		  const isExport = type === 'export';
-		  if(isExport || type === 'display') {
-			if(data == 'Z')
-			  return isExport ? 'Zakwalifikowany' : '<span class="qualified">Zakwalifikowany</span>';
-			else if(data == "None")
-			  return '';
-			else if(data == 'O')
-			  return isExport ? 'Odrzucony' : '<span class="not-qualified">Odrzucony</span>' ;
-			else
-			  return data;
-		  } else
-			return data;
-		}
-	  },
-	  {
-		targets: 'datatables-yes-no',
-		render: (data, type, row) => {
-		  if(type === 'export') {
-			return data.split("> ").pop();
-		  } else
-			return data;
-		}
-	  }
-	],
-	buttons: [
-		{
-			extend: 'colvis',
-			columns: ':gt(0)'
-		},
-		{
-			extend: 'copy',
-			exportOptions: {
-				columns: column_selector,
-				orthogonal: 'export'
-			}
-		},
-		{
-			extend: 'excel',
-			exportOptions: {
-				columns: column_selector,
-				orthogonal: 'export'
-			}
-		},
-		{
-			extend: 'pdf',
-			exportOptions: {
-				columns: column_selector,
-				orthogonal: 'export'
-			},
-	   },
-	   {
-			extend: 'print',
-			exportOptions: {
-				columns: column_selector,
-				orthogonal: 'export'
-			}
-	   },
-	],
-	"language": {
-	  "url": datatables_polish_locale_url,
-	},
-	"fnRowCallback" : function(nRow, aData, iDisplayIndex){
-	  $("td:first", nRow).html(iDisplayIndex +1);
-	  return nRow;
-	},
-	"pageLength": 50,
-	"lengthMenu": [[10, 25, 50, 100, -1], [10, 25, 50, 100, "All"]],
+    // https://datatables.net/forums/discussion/42192/exporting-data-with-buttons-and-responsive-extensions-controlled-by-column-visibility
+    if ($(node).hasClass('noVis')) {
+      return false;
+    }
+    const tableID = $(node).closest('table').attr('id');
```

Comment:
You are getting the id attribute from the node in order to get the node by id in the next line? :joy:

Reply:
Yes, somehow it didn't work. Will test it again to be absolutely sure.

Reply:
What if the node does not have an ID like the workshops table? It surely doesn't work in that case, unless something autogenerates an ID and adds it after the fact

Reply:
Datatables autogenerates an ID if the table doesn't already have an id.

Reply:
I've rechecked, when you use the table obtained by the closest function then all columns are being exported regardless of visibility status. Checking if we got a null/undefined table still doesn't work.

Reply:
But you did change it in 30089403ef29ab660a63950b0d35ade54a39af1f, does that work or not?!

Reply:
It works, I thought when jquery doesn't find an element it returns and undefined/null element instead of an empty list.